hooks/mutations/delegation/useIgnoreReceivedRequestToBeDelegate.ts

```diff
@@ -0,0 +1,54 @@
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { ignoredRequestToBeDelegateAddressesKey } from "constants/queryKeys";
+import { getIgnoredRequestToBeDelegateAddressesFromStorage } from "helpers";
+import { useHandleError, useUserContext } from "hooks";
+
+export function useIgnoreReceivedRequestToBeDelegate() {
+  const { address } = useUserContext();
+  const onError = useHandleError();
+  const queryClient = useQueryClient();
+
+  const { mutate, isLoading } = useMutation(ignoreReceivedRequestToBeDelegate, {
+    onError,
+    onSuccess: (_data, { delegatorAddress }) => {
+      queryClient.setQueryData<string[]>(
+        [ignoredRequestToBeDelegateAddressesKey, address],
+        (oldIgnoredRequestToBeDelegateAddresses) => {
+          if (!oldIgnoredRequestToBeDelegateAddresses) return;
+
+          const newIgnoredRequestToBeDelegateAddresses = [...oldIgnoredRequestToBeDelegateAddresses, delegatorAddress];
+
+          return newIgnoredRequestToBeDelegateAddresses;
+        }
+      );
+    },
+  });
+
+  return {
+    ignoreReceivedRequestToBeDelegateMutation: mutate,
+    isIgnoringRequestToBeDelegate: isLoading,
+  };
+}
+
+function ignoreReceivedRequestToBeDelegate({
+  userAddress,
+  delegatorAddress,
+}: {
+  userAddress: string;
+  delegatorAddress: string;
+}) {
+  return new Promise(() => {
```

Comment:
react query only works with functions that return promises, so this is wrapped in a promise even though it is not an async operation

Reply:
why not make the function `ignoreReceivedRequestToBeDelegate` async. also i dont think this promise works right because its never resolved or rejected

Reply:
I actually did it like that at first but then the typescript eslint complained. But yeah thinking of it again I prefer just making it async and I'll add an ignore there.